src/deps_log.cc

```diff
@@ -218,9 +228,16 @@ bool DepsLog::Load(const string& path, State* state, string* err) {
       assert(size % 4 == 0);
       int* deps_data = reinterpret_cast<int*>(buf);
       int out_id = deps_data[0];
-      int mtime = deps_data[1];
-      deps_data += 2;
-      int deps_count = (size / 4) - 2;
+      TimeStamp mtime;
+      if (little_endian) {
+        mtime = (TimeStamp)((uint64_t)deps_data[2] << 32 |
+                            (uint64_t)deps_data[1]);
+      } else {
+        mtime = (TimeStamp)((uint64_t)deps_data[1] << 32 |
+                            (uint64_t)deps_data[2]);
+      }
```

Comment:
Wouldn't casting to uint64_t* and deference, instead of having to handle endianess be good enough? Does anyone really care builds on platforms with misalignment access issues?

Also, an alternative could be writing mtime in two parts (low32, high32), so you could combine these in the same way w/o considering platform endianess.